## (a)

In [2]:
set.seed(1)
x = rnorm(100)
y = 2*x + rnorm(100)

lm.fit = lm(y~x+0)

summary(lm.fit)


Call:
lm(formula = y ~ x + 0)

Residuals:
    Min      1Q  Median      3Q     Max 
-1.9154 -0.6472 -0.1771  0.5056  2.3109 

Coefficients:
  Estimate Std. Error t value Pr(>|t|)    
x   1.9939     0.1065   18.73   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.9586 on 99 degrees of freedom
Multiple R-squared:  0.7798,	Adjusted R-squared:  0.7776 
F-statistic: 350.7 on 1 and 99 DF,  p-value: < 2.2e-16


$\beta = 1.9939$ with a $SE = 0.1065$

From the F-statistic and its respcetive p-value we can infer that the variable explains a lot of the variability in $Y$.

$X$ has significantly low p-value, therefore we can reject the null hypothesis

## (b)

In [3]:
lm.fit2 = lm(x~y+0)
summary(lm.fit2)


Call:
lm(formula = x ~ y + 0)

Residuals:
    Min      1Q  Median      3Q     Max 
-0.8699 -0.2368  0.1030  0.2858  0.8938 

Coefficients:
  Estimate Std. Error t value Pr(>|t|)    
y  0.39111    0.02089   18.73   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.4246 on 99 degrees of freedom
Multiple R-squared:  0.7798,	Adjusted R-squared:  0.7776 
F-statistic: 350.7 on 1 and 99 DF,  p-value: < 2.2e-16


$\beta = 0.3911$ with $SE = 0.02089$
We get identical values for the F-statistic, t-statistic, and p-values.
Therefore we can reject the null hypothesis.

## (c)
Both results refer to the same line.

## (d)

$$t=\frac{\hat{\beta}}{SE(\hat{\beta})}$$
$$SE(\beta)=\sqrt{\frac{\sum_{i=1}^{n}(y_i-x_i\hat{\beta})^2}{(n-1)\sum_{i'=1}^{n}x_{i'}^2}}$$

Remember that for this simple linear regression scenario we estimate the slope by:
$$\hat{\beta_1} = \frac{\sum_{i=1}^{n} x_iy_i}{\sum_{i=1}^{n}x_i^2}$$


$$t=\frac{\frac{\sum_{i=1}^{n} x_iy_i}{\sum_{i=1}^{n}x_i^2}}{\sqrt{\frac{\sum_{i=1}^{n}(y_i-x_i\hat{\beta})^2}{(n-1)\sum_{i'=1}^{n}x_{i'}^2}}}$$


$$t={\frac{\sum_{i=1}^{n} x_iy_i}{\sum_{i=1}^{n}x_i^2}}\sqrt{\frac{(n-1)\sum_{i'=1}^{n}x_{i'}^2}{\sum_{i=1}^{n}(y_i-x_i\hat{\beta})^2}}$$


$$t=\frac{(\sum_{i=1}^{n} x_iy_i)\sqrt{n-1}}{\sqrt{\sum_{i=1}^{n}x_i^2\sum_{i=1}^{n}(y_i-x_i\hat{\beta})^2}}$$


$$t=\frac{(\sum_{i=1}^{n} x_iy_i)\sqrt{n-1}}{\sqrt{\sum_{i=1}^{n}x_i^2\sum_{i=1}^{n}(y_i^2-2y_ix_i\hat{\beta}+(x_i\hat{\beta})^2)}}$$


$$t=\frac{(\sum_{i=1}^{n} x_iy_i)\sqrt{n-1}}{\sqrt{\sum_{i=1}^{n}x_i^2\sum_{i=1}^{n}y_i^2-\sum_{i=1}^{n}(2y_ix_i\hat{\beta}+x_i^2\hat{\beta^2})}}$$


*I'm stuck here...*

$$t=\frac{(\sum_{i=1}^{n} x_iy_i)\sqrt{n-1}}{\sqrt{\sum_{i=1}^{n}x_i^2\sum_{i=1}^{n}y_i^2-\sum_{i=1}^{n}(x_iy_i)^2}}$$



In [7]:
(sum(x*y) * sqrt(length(x)-1)) / (sqrt(sum(x^2)*sum(y^2)-(sum(x*y))^2))

[1] 18.72593

Same value as the t-statistic computed before.

## (e)

Swaping x and y will yield the same result

## (f)

In [10]:
lm.fit3 = lm(x~y)
summary(lm.fit3)

lm.fit4 = lm(y~x)
summary(lm.fit4)


Call:
lm(formula = x ~ y)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.90848 -0.28101  0.06274  0.24570  0.85736 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  0.03880    0.04266    0.91    0.365    
y            0.38942    0.02099   18.56   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.4249 on 98 degrees of freedom
Multiple R-squared:  0.7784,	Adjusted R-squared:  0.7762 
F-statistic: 344.3 on 1 and 98 DF,  p-value: < 2.2e-16



Call:
lm(formula = y ~ x)

Residuals:
    Min      1Q  Median      3Q     Max 
-1.8768 -0.6138 -0.1395  0.5394  2.3462 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) -0.03769    0.09699  -0.389    0.698    
x            1.99894    0.10773  18.556   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.9628 on 98 degrees of freedom
Multiple R-squared:  0.7784,	Adjusted R-squared:  0.7762 
F-statistic: 344.3 on 1 and 98 DF,  p-value: < 2.2e-16
